### PR TITLE
Output viscosity to pvd file

### PR DESCRIPTION
**Pull request by CaroleAnneDaunais**
When a rheological model is used for a non Newtonian flow, the viscosity is output in the pvd file.

### CODE REVIEW

File: include/solvers/post_processors.h [hunk 1]
```diff
@@ -166,4 +170,57 @@ class SRFPostprocessor : public DataPostprocessorVector<dim>
...
+    Tensor<2, dim>     shear_rate;
+    const unsigned int n_quadrature_points = inputs.solution_gradients.size();
+
+    std::shared_ptr<RheologicalModel<dim>> rheological_model;
+    if (non_newtonian_parameters.model ==
+        Parameters::NonNewtonian::Model::carreau)
+      {
+        rheological_model =
+          std::make_shared<Carreau<dim>>(non_newtonian_parameters);
+      }
```
Comment (blaisb) [inline include/solvers/post_processors.h:199]: I would create the rheological model in the constructor, not in the call to evaluate vector field.

File: source/solvers/navier_stokes_base.cc [hunk 2]
```diff
@@ -49,6 +49,8 @@
 #include <deal.II/opencascade/manifold_lib.h>
 #include <deal.II/opencascade/utilities.h>
 
+#include <boost/lexical_cast.hpp>
```
Comment (blaisb) [inline source/solvers/navier_stokes_base.cc:52]: Why do you need this?

File: source/solvers/navier_stokes_base.cc [hunk 3]
```diff
@@ -1526,6 +1528,14 @@ NavierStokesBase<dim, VectorType, DofsType>::write_output_results(
       Parameters::VelocitySource::VelocitySourceType::srf)
     data_out.add_data_vector(solution, srf);
 
+  NonNewtonianViscosityPostprocessor<dim> non_newtonian_viscosity(
+    simulation_parameters.physical_properties.non_newtonian_parameters);
+  if (simulation_parameters.physical_properties.non_newtonian_flow)
+    {
+      data_out.add_data_vector(solution, non_newtonian_viscosity);
+    }
```
Comment (blaisb) [inline source/solvers/navier_stokes_base.cc:1536]:
```suggestion
      data_out.add_data_vector(solution, non_newtonian_viscosity);
```

**Review by blaisb — COMMENTED** [review]
Minor comments / easy fixes

File: source/solvers/navier_stokes_base.cc (hunk 2 above)
Comment (CaroleAnneDaunais) [reply source/solvers/navier_stokes_base.cc:52]: I don't need it, my mistake

File: include/solvers/post_processors.h (hunk 1 above)
Comment (CaroleAnneDaunais) [reply include/solvers/post_processors.h]: I changed it.

**Pull request merged**